### ISSUE TITLE
unicorn/no-fn-reference-in-iterator - wrong context

### ISSUE THREAD

**Issue by lagden** [sindresorhus/eslint-plugin-unicorn#755]
The lint is claiming a MongoDB query (`.find`)

```js
// Example
// error  Do not pass function `collection` directly to `.find(…)`  unicorn/no-fn-reference-in-iterator
const results = await collection
  .find({
    $and: [cursorQuery, params.query]
  }, {
    projection: params.projection
  })
  .sort($sort)
  .limit(params.limit + 1)
  .toArray()
```

**Comment by fisker** [comment]
I can make a PR to  ignore objects for now, but

```js
const results = await collection
  .find(
   foo,  // <-- we  can't tell  what this  is
    {
    projection: params.projection
  })
  .sort($sort)
  .limit(params.limit + 1)
  .toArray()
```

still  get reported.

**Comment by fisker** [comment]
This message in issue description might misleading, the message is actually

```text
error  Do not pass function directly to `.find(…)`  unicorn/no-fn-reference-in-iterator
```

**Comment by lagden** [comment]
The message is from `eslint`.

![image](https://user-images.githubusercontent.com/130963/83292839-31885600-a1c1-11ea-9d76-9cb92eac96b0.png)

**Comment by fisker** [comment]
```diff
- error  Do not pass function `collection` directly to `.find(…)`  unicorn/no-fn-reference-in-iterator
+ error  Do not pass function directly to `.find(…)`  unicorn/no-fn-reference-in-iterator
```
there is no function name in that message.

**Comment by lagden** [comment]
@fisker sorry!!

I copied error from other block!! my bad!

```js
// error  eslint:unicorn/no-fn-reference-in-iterator  Do not pass function `collection` directly to `.find(…)`.
const collection = t.context.db.collection('test_paging')
let res = await paging.find(collection, {
  limit: 3
})

**Comment by fisker** [comment]
No problem. As I comment above, if you use object in `.find` this will ignored in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/756. But `.find(foo)` will still be a problem, you have to use `//eslint-disable`.

**Comment by aleksandr-omise** [comment]
With _D3 delaunay_ error too.
```
const delaunay = Delaunay.from(
  valuesWithCoord,
  (el) => el.x,
  (el) => el.y
)

const nearestPointIndex = delaunay.find(x, y)
```
I guess nothing can do about it.
